### PR TITLE
[CALCITE] Added support for host variables

### PR DESCRIPTION
**Pull request by michaelvdow**
Host variables are variables prefixed with a colon (`:variableName`)

### CODE REVIEW

File: babel/src/test/java/org/apache/calcite/test/BabelParserTest.java [hunk 1]
```diff
@@ -2255,4 +2255,51 @@ private void checkParseInfixCast(String sqlType) {
...
+
+  @Test public void testHostVariableExecNamedParams() {
+    final String sql = "exec foo (bar=:bar, baz=:baz, qux=:qux)";
+    final String expected = "EXECUTE `FOO` (`BAR` = :BAR, `BAZ` = :BAZ,"
+        + " `QUX` = :QUX)";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testHostVariableSelect() {
+    final String sql = "select :bar from foo where a = :qux";
```
Comment (dasch-1) [inline babel/src/test/java/org/apache/calcite/test/BabelParserTest.java:2273]: nit: Can we add an alias after ":bar" just to cover that case too?

File: babel/src/main/java/org/apache/calcite/sql/babel/SqlHostVariable.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+   * @param pos  Parser position, must not be null
+   */
+  public SqlHostVariable(String name, SqlParserPos pos) {
+    super(name, pos);
+  }
+
+  @Override public void unparse(SqlWriter writer, int leftPrec, int rightPrec) {
+    writer.print(":");
+    writer.print(names.get(0));
+    // Ensures whitespace is added before a separator such as "AS"
```
Comment (AndrewPochapsky) [inline babel/src/main/java/org/apache/calcite/sql/babel/SqlHostVariable.java:41]: nit: punctuation

File: babel/src/main/java/org/apache/calcite/sql/babel/SqlHostVariable.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+   * @param name  Name of the host variable
+   * @param pos  Parser position, must not be null
+   */
+  public SqlHostVariable(String name, SqlParserPos pos) {
+    super(name, pos);
+  }
+
+  @Override public void unparse(SqlWriter writer, int leftPrec, int rightPrec) {
+    writer.print(":");
+    writer.print(names.get(0));
```
Comment (AndrewPochapsky) [inline babel/src/main/java/org/apache/calcite/sql/babel/SqlHostVariable.java:40]: nit: Can these two lines be combined into one?

File: babel/src/test/java/org/apache/calcite/test/BabelParserTest.java (hunk 1 above)
Comment (michaelvdow) [reply babel/src/test/java/org/apache/calcite/test/BabelParserTest.java:2273]: Sure I can add that

File: babel/src/main/java/org/apache/calcite/sql/babel/SqlHostVariable.java (hunk 3 above)
Comment (michaelvdow) [reply babel/src/main/java/org/apache/calcite/sql/babel/SqlHostVariable.java]: Yeah good point, I updated it

File: babel/src/main/java/org/apache/calcite/sql/babel/SqlHostVariable.java (hunk 2 above)
Comment (michaelvdow) [reply babel/src/main/java/org/apache/calcite/sql/babel/SqlHostVariable.java]: Fixed

**Pull request merged**